### PR TITLE
Update to version 0.25

### PR DESCRIPTION
**Pull request by Tomin1**
Update submodule and refresh patches. Dropped those that are no longer relevant. Removed qt4 packaging.

### CODE REVIEW

File: rpm/0005-Support-OneDrive-oauth2-flow.patch [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+-                } else if (pair.first == EXPIRES_IN) {
+-                    respData.setExpiresIn(pair.second.toInt());
+-                } else if (pair.first == REFRESH_TOKEN) {
+-                    respData.setRefreshToken(pair.second);
+-                } else if (pair.first == STATE) {
+-                    state = pair.second;
+-                } else if (pair.first == SCOPE) {
+-                    respData.setScope(pair.second.split(' ', QString::SkipEmptyParts));
++        // some providers (e.g. OneDrive) implement non-standard OAuth2 flows
++        // and return auth code requiring exchange for token.
```
Comment (Tomin1) [inline rpm/0005-Support-OneDrive-oauth2-flow.patch:58]: I'm just wondering if this should now add `state` into the url. Previously `state` was set by signonui. Or if possible we'd drop this patch but I don't know if OneDrive still needs this.

File: rpm/signon-plugin-oauth2-qt5.spec [hunk 2]
```diff
@@ -1,26 +1,22 @@
...
+URL: https://github.com/sailfishos/signon-plugin-oauth2
 Source0: %{name}-%{version}.tar.bz2
 Patch0: 0001-Manually-time-out-HTTP-requests-after-30-seconds.patch
 Patch1: 0002-OAuth2-Relax-RefreshToken-restriction-on-ProvidedTok.patch
 Patch2: 0003-Always-install-to-usr-lib-never-usr-lib64.patch
 Patch3: 0004-Always-force-client-auth-via-request-body.patch
-Patch4: 0005-Skip-unstable-unit-tests.patch
-Patch5: 0006-Fix-dropbox-authentication-by-removing-type-query-ar.patch
-Patch6: 0007-Support-OneDrive-oauth2-flow.patch
+Patch6: 0005-Support-OneDrive-oauth2-flow.patch
```
Comment (vigejolla) [inline rpm/signon-plugin-oauth2-qt5.spec:12]: Should this be now Patch4?

**Review by vigejolla — COMMENTED** [review]
If we ignore the OneDrive issue, this looks fine to me. So, I guess we need one brave soul to try this with OneDrive?

File: rpm/0005-Support-OneDrive-oauth2-flow.patch (hunk 1 above)
Comment (vigejolla) [reply rpm/0005-Support-OneDrive-oauth2-flow.patch:58]: Looking more carefully, I'm confident that this works like it should, assuming the old version used to work. Just the question if we could drop the patch remains.

**Review by vigejolla — APPROVED** [review]
LGTM, I suggest creating a follow-up task for checking if the OneDrive patch could be dropped.

**Pull request merged**